src/server.cpp

```diff
@@ -150,50 +159,91 @@ int main () {
+    std::shared_ptr<HSMP::ClientRequest> request, User &current_user) {
 
   switch (request->type()) {
 
     case HSMP::RequestType::kLoginRequest: {
-
+      auto lreq = std::static_pointer_cast<HSMP::LoginRequest>(request);
       auto lres = std::make_shared<HSMP::LoginResponse>();
-
-      return lres;
+      if (current_user.GetName() == lreq->user) {
+        if (current_user.GetPassword() == lreq->passwd) {
+          current_user.SetOnline();
+          lres->ok = "ok";
+          return lres;
+        }
+        else {
+          auto eres = std::make_shared<HSMP::ErrorResponse>();
+          eres->message = kMessageErrorPassword;
+          return eres;
+        }
+      }
+      auto eres = std::make_shared<HSMP::ErrorResponse>();
+      eres->message = kMessageErrorPersonDontExist;
+      return eres;
     }
 
     case HSMP::RequestType::kListaRequest: {
       auto ires = std::make_shared<HSMP::ListaResponse>();
-
+      for (User &listed_user : *users) {
+        if (listed_user.IsOnline()) {
+          ires->tam_user_names.push_back(listed_user.GetName().size());
+          ires->user_names.push_back(listed_user.GetName());
+        }
+      }
+      ires->num_users = ires->user_names.size();
       return ires;
     }
 
     case HSMP::RequestType::kMessageRequest: {
+      // Se asume que las comprobaciones se hacen en AttendConnection para saber
+      // a quien se debe mandar el response, y si ese usuario existe y esta conectado
+      auto mreq = std::static_pointer_cast<HSMP::MessageRequest>(request);
       auto mres = std::make_shared<HSMP::MessageResponse>();
-
+      mres->tam_msg = mreq->tam_msg;
+      mres->msg = mreq->msg;
+      mres->tam_remitente = current_user.GetName().size();
+      mres->remitente = current_user.GetName();
       return mres;
     }
 
     case HSMP::RequestType::kBroadcastRequest: {
+      // Se asume que el mismo mensaje se manda a todos los usuarios conectados
+      auto breq = std::static_pointer_cast<HSMP::BroadcastRequest>(request);
       auto bres = std::make_shared<HSMP::BroadcastResponse>();
-      // write(ConnectFD, res->ParseToCharBuffer(), Klenght);
+      bres->tam_msg = breq->tam_msg;
+      bres->msg = breq->msg;
+      bres->tam_remitente = current_user.GetName().size();
+      bres->remitente = current_user.GetName();
       return bres;
     }
 
     case HSMP::RequestType::kUploadFileRequest: {
+      auto ureq = std::static_pointer_cast<HSMP::UploadFileRequest>(request);
       auto ures = std::make_shared<HSMP::UploadFileResponse>();
-
+      ures->tam_file_name = ureq->tam_file_name;
+      ures->tam_file_data = ureq->tam_file_data;
+      ures->tam_remitente = current_user.GetName().size();
+      ures->file_name = ureq->file_name;
+      ures->remitente = current_user.GetName();
+      strcpy(ures->file_data, ureq->file_data);
       return ures;
     }
 
+      // UploadRequest deberia devolver un File_ANResponse al destinatario
```

Comment:
this mean:
Miguel send UpLoadFileRequest-Joaquin -> Server
Server send ANRequest-Miguel -> Joaquin ?
Joaquin send ANRequest-.... -> Server ?
....... ?
Honestly, I don't understand how traceability is... ANRequest contain only names, how write accept o reject?

Reply:
Debería cambiarse el protocolo, porque como está ahora no tiene sentido la interacción. Por el momento creo que debería llegar el response de upload sin el ANrequest y hacer la verificación de si quiere o no recibir el archivo localmente, es decir, del lado del propio cliente.